include/boost/geometry/extensions/triangulation/strategies/cartesian/detail/precise_math.hpp

```diff
@@ -376,6 +314,47 @@ inline RealNumber orient2d(std::array<RealNumber, 2> const& p1,
     return(D[D_nz - 1]);
 }
 
+// see page 38, Figure 21 for the calculations, notation follows the notation in the figure.
+template
+<
+    typename RealNumber,
+    std::size_t Robustness = 3
+>
+inline RealNumber orient2d(std::array<RealNumber, 2> const& p1,
+                           std::array<RealNumber, 2> const& p2,
+                           std::array<RealNumber, 2> const& p3)
+{
+    if(Robustness == 0) {
+        return (p1[0]-p3[0])*(p2[1]-p3[1]) - (p1[1]-p3[1])*(p2[0] - p3[0]);
+    }
+    std::array<RealNumber, 2> t1, t2, t3, t4;
```

Comment:
Using this means that we can merge it only in `1.75` as C++03 is then gone.
Alternatively you can use `boost::array` - but I don't know if more of your code is dependent on C++11

Reply:
Would that establish an interface or would later breaking that interface (to use std::array again) be ok because it is in the detail namespace?

I would be fine if it lives in develop/extensions until 1.75 next year. I could also leave it this way and revisit the option to change it to `boost::array` only, if the inclusion of side_robust into the master branch will be considered before 1.74.

Reply:
Would it make sense to replace `array` with your own clearly named type? It's a 2d vector correct? If that's the case then why not call it like that (e.g. `vec2d<>`) and have `x` and `y` members or use `boost::model::point_xy<>`?

Reply:
Thanks for the suggestion with regard to the interface, it is true that the values taken there correspond to points, so having points as parameters would be sensible. The matter of the remaining std::arrays would no longer be a matter of the interface of the function.

I care a lot about the performance of those functions, though. Judging from the source code, I guess that point_xy and boost::array are zero-overhead like std::array but I'll verify that in the next couple of days via benchmarks/looking at the assembly and then I'll commit the necessary changes.

Reply:
@awulkiew I replaced the arrays for p1, p2, p3 with an own type (named vec2d). All other arrays in the function are expansions of single values and not 2d or nd vectors, so I did not touch those. Out of your two suggestions (own type and point_xy), I opted for my own type because I noticed that point_xy is not included almost nowhere (except in geometry/strategies/geographic/distance.hpp ), so I didn't want to create a header that depends on it.

I'm fine with your general idea about not including side_robust anywhere by default before 1.75, So I did not change anything with regard to std::array and std::fma.